### PR TITLE
fix: avoid create pluginconfig in the tranlsation of route (#836)

### PR DESCRIPTION
**Pull request by neverCase**
### Type of change:

- [X] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by neverCase** [comment]
ci need re-run @tao12345666333

**Review by tokers — COMMENTED** [review]
Need some test cases to make sure the plugin_config_id is not written now.

**Comment by neverCase** [comment]
> Need some test cases to make sure the plugin_config_id is not written now.

ok, I will add more test cases today.

**Comment by neverCase** [comment]
e2e test need to re-run @tao12345666333

**Comment by neverCase** [comment]
ci error below:
```sh
• Failure [70.707 seconds]
ApisixRoute Testing
/home/runner/work/apisix-ingress-controller/apisix-ingress-controller/test/e2e/ingress/resourcepushing.go:28
  service is referenced by two ApisixRoutes [It]
  /home/runner/work/apisix-ingress-controller/apisix-ingress-controller/test/e2e/ingress/resourcepushing.go:502

  
  	Error Trace:	resourcepushing.go:607
  	            				runner.go:113
  	            				runner.go:64
  	            				it_node.go:26
  	            				spec.go:215
  	            				spec.go:138
  	            				spec_runner.go:200
  	            				spec_runner.go:170
  	            				spec_runner.go:66
  	            				suite.go:79
  	            				ginkgo_dsl.go:238
  	            				ginkgo_dsl.go:213
  	            				e2e_test.go:26
  	Error:      	"[%!s(*v1.Upstream=&{{45e6c8e6 ingress-apisix-e2e-tests-default-428202328_httpbin-service-e2e-test_80 Created by apisix-ingress-controller, DO NOT modify it manually map[managed-by:apisix-ingress-controller]} roundrobin vars  <nil> [{10.244.2.32 80 100}] http <nil> <nil> <nil>})]" should have 0 item(s), but has 1
  	Test:       	ApisixRoute Testing service is referenced by two ApisixRoutes
```
maybe it need to re-run once again. @tao12345666333

**Comment by tao12345666333** [comment]
Wait for #859 to fix CI.

**Comment by tao12345666333** [comment]
https://github.com/apache/apisix-ingress-controller/pull/859 has been merged. Re-run the CI.

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Comment by tao12345666333** [comment]
ping @tokers @gxthrj  for review, thanks

File: pkg/kube/translation/apisix_route.go
Comment (tokers) [inline pkg/kube/translation/apisix_route.go:534]: We don't use the plugin config name as the id, see https://github.com/apache/apisix-ingress-controller/blob/master/pkg/kube/translation/apisix_pluginconfig.go#L51 for details.

Comment (neverCase) [reply pkg/kube/translation/apisix_route.go:534]: ok, I will refactor it right now

**Comment by neverCase** [comment]
Actually, I found I had covert the plugin config name to plugin id in the apisixRouteController.
https://github.com/apache/apisix-ingress-controller/blob/master/pkg/ingress/apisix_route.go#L240

**Comment by neverCase** [comment]
It's been a long time and I almost forgot what I had done

File: pkg/kube/translation/apisix_route.go
Comment (tokers) [inline pkg/kube/translation/apisix_route.go:980]: Why we don't handle the `PluginConfigName` in this method?

Comment (neverCase) [reply pkg/kube/translation/apisix_route.go:980]: That's because when it creates the ApisixRoute and the field `PluginConfigName` has been specified, I want to use apisix.Cluster api to check the plugin was whether exist.
 And at the same time replace it with the plugin id.
...
Since the method to generate the plugin id is fixed.
I will just remove the replacement and remain checking.

File: pkg/ingress/apisix_route.go
Comment (tokers) [inline pkg/ingress/apisix_route.go:248]: Should distinguish the error type, not all errors are "not found".

File: pkg/kube/translation/apisix_route.go
Comment (tokers) [reply pkg/kube/translation/apisix_route.go:980]: Well, I mean `translateHTTPRouteV2beta3NotStrictly` is basically same to `translateHTTPRouteV2beta3`, so here you should also set the plugin config id.

File: pkg/ingress/apisix_route.go
Comment (neverCase) [reply pkg/ingress/apisix_route.go:248]: ok

**Comment by neverCase** [comment]
@tao12345666333 
The case `check the ingress lb status is updated` ran failed.
Please re-run.

**Comment by neverCase** [comment]
@tao12345666333 
Ingress case `check the ingress lb status is updated` run failed.

**Comment by tao12345666333** [comment]
let me re-run it

**Review by gxthrj — APPROVED** [review]
LGTM

**Comment by tao12345666333** [comment]
CI is green. Wait for @tokers  approval

**Review by tokers — APPROVED** [review]

**Pull request merged**
